### PR TITLE
LZW compression algorithm

### PR DESCRIPTION
**Pull request by Quentin61**
### **Describe your change:**

I've created an algorithm to compress and decompresse strings with the LZW method. The compressed string is formated in ASCII code separated into '_'.

* [x] Add an algorithm?
* [ ] ~~Fix a bug or typo in an existing algorithm?~~
* [ ] ~~Documentation change?~~

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

**This PR is NOT ready for review**

### CODE REVIEW

File: compression/lzw.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from src.dictionary import Dictionary
```
Comment (cclauss) [inline compression/lzw.py]: from .dictionary import Dictionary

File: compression/lzw_main.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from src.lzw import LZW
```
Comment (cclauss) [inline compression/lzw_main.py]: from .lzw import LZW

**Comment by cclauss** [comment]
OK...  Now we pass the Travis CI tests.  We need to add some type annotations and doctests.

File: compression/lzw_dictionary.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class Dictionary:
+    def __init__(self):
+        """
+        Dictionary constructor
+        """
```
Comment (cclauss) [inline compression/lzw_dictionary.py]: Please add doctests like this...
```
class Dictionary:
     def __init__(self):
         """
         Dictionary constructor

        >>> d = Dictionary()
        >>> d.search("?")
        -1
        >>> d.search("")
        -2
        >>> d.print_dictionary()

        >>> d.add_to_dictionary("A")
        >>> d.search("?")
        -1
        >>> d.search("")
        -2
        >>> d.search("A")
        'A'
        >>> d.print_dictionary()
        'A'
        """
```

**Comment by onlinejudge95** [comment]
@Quentin61 I would like to see this in our collection, please reach out in case of any help required

**Pull request closed without merging**